l2tdevtools/build_helpers/wheel.py

```diff
@@ -0,0 +1,236 @@
+
+    Args:
+      source_helper_object (SourceHelper): source helper.
+    """
+    # Remove previous versions of wheels.
+    project_name, project_version = self._GetWheelFilenameProjectInformation(
+        source_helper_object)
+
+    filenames_to_ignore = '{0:s}-{1:s}-.*-.*-.*.whl'.format(
+        project_name, project_version)
+    filenames_to_ignore = re.compile(filenames_to_ignore)
+
+    filenames_glob = '{0:s}-*-*-*-*.whl'.format(project_name)
+    filenames = glob.glob(filenames_glob)
+
+    for filename in filenames:
+      if not filenames_to_ignore.match(filename):
+        logging.info('Removing: {0:s}'.format(filename))
+        os.remove(filename)
+
+
+class ConfigureMakeWheelBuildHelper(WheelBuildHelper):
+  """Helper to build Python wheel packages (.whl)."""
+
+  def Build(self, source_helper_object):
+    """Builds the wheel.
+
+    Args:
+      source_helper_object (SourceHelper): source helper.
+
+    Returns:
+      bool: True if successful, False otherwise.
+
+    Raises:
+      RuntimeError: if setup.py is missing and a wheel cannot be build.
+    """
+    source_filename = source_helper_object.GetSourcePackageFilename()
+    if not source_filename:
+      logging.info('Missing source package of: {0:s}'.format(
+          source_helper_object.project_name))
+      return False
+
+    source_directory = source_helper_object.GetSourceDirectoryPath()
+    if not source_directory:
+      logging.info('Missing source directory of: {0:s}'.format(
+          source_helper_object.project_name))
+      return False
+
+    logging.info('Building wheel of: {0:s}'.format(source_filename))
+
+    setup_py_path = os.path.join(source_directory, 'setup.py')
+    if not os.path.exists(setup_py_path):
+      raise RuntimeError('Missing setup.py cannot build wheel')
+
+    log_file_path = os.path.join('..', self.LOG_FILENAME)
+    command = '\"{0:s}\" setup.py bdist_wheel > {1:s} 2>&1'.format(
+        sys.executable, log_file_path)
+    exit_code = subprocess.call('(cd {0:s} && {1:s})'.format(
+        source_directory, command), shell=True)
+    if exit_code != 0:
+      logging.error('Running: "{0:s}" failed.'.format(command))
+      return False
+
+    return self._MoveWheel(source_helper_object)
+
+
+class SetupPyWheelBuildHelper(WheelBuildHelper):
+  """Helper to build Python wheel packages (.whl)."""
+
+  def Build(self, source_helper_object):
+    """Builds the wheel.
+
+    Args:
+      source_helper_object (SourceHelper): source helper.
+
+    Returns:
+      bool: True if successful, False otherwise.
+    """
+    source_filename = source_helper_object.GetSourcePackageFilename()
+    if not source_filename:
+      logging.info('Missing source package of: {0:s}'.format(
+          source_helper_object.project_name))
+      return False
+
+    source_directory = source_helper_object.GetSourceDirectoryPath()
+    if not source_directory:
+      logging.info('Missing source directory of: {0:s}'.format(
```

Comment:
Please add the source directory to this error message to assist in debugging.

Reply:
`source_directory` is empty, how should it be added to the string?